### ISSUE TITLE
Should `toStrict` return a `Strict` entity?

### ISSUE THREAD

**Issue by armanbilge** [http4s/http4s#6516]
It's currently a bit weird that it doesn't.

https://github.com/http4s/http4s/blob/6b74773babb2b4ae31aeb3c09056acb9326ab762/core/shared/src/main/scala/org/http4s/Message.scala#L255-L257

OTOH, it's a bit weird how to handle the empty entity case if it did.

**Comment by danicheg** [comment]
This method returns an updated `Message` that contains a method `entity: Entity`. So, if I understand correctly, you're confused with the `Entity.Empty` case. I think the sense of the 'strict' from the method name is that the whole message entity will be drained into memory. From that perspective, the `Entity.Empty` is also 'strict'. FWIW (considering returning type of the method): tastes differ, but exposing the concrete subtype of ADT within the API contract — it smells as for me.

**Comment by danicheg** [comment]
Yeah, having this method named `toStrict` when we have the `Entity` model is confusing a little. :shrug: I can advise only adding the `toChunk` method to `Entity`. Then you could flatmap things nicely; + in runtime, it will be performed on the same thread; + we will hide redundant machinery because of the type system at the user side.

**Comment by bplommer** [comment]
One possibility would be to expose the `Entity` subtype in `Message` as a type member, then introduce a supertype of `Entity.Strict` and `Entity.Empty` - then `toStrict` could return something like `F[Request.Aux[Pure, Entity.StrictOrEmpty]]` - I'm reasonably optimistic this could be made to work, but not sure how useful it would be in practice?
